test/jdk/jdk/internal/util/VersionTest.java

```diff
@@ -0,0 +1,108 @@
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.fail;
+
+/**
+ * @test
+ * @summary test jdk.internal.util.Version
+ * @modules java.base/jdk.internal.util
+ * @run junit VersionTest
+ */
+
+public class VersionTest {
+
+    private static Stream<Arguments> versionParams() {
+        return Stream.of(
+                Arguments.of("1", new Version(1, 0)),
+                Arguments.of("1.2", new Version(1, 2)),
+                Arguments.of("1.2", new Version(1, 2, 0)),
+                Arguments.of("1.2.3", new Version(1, 2, 3)),
+                Arguments.of("1-abc", new Version(1, 0, 0)), // Ignore extra
+                Arguments.of("1.2-abc", new Version(1, 2, 0)), // Ignore extra
+                Arguments.of("1.2.3.4", new Version(1, 2, 3)), // Ignore extra
+                Arguments.of("1.2.3-abc", new Version(1, 2, 3)) // Ignore extra
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("versionParams")
+    public void checkParse(String verName, Version expected) {
+        Version actual = Version.parse(verName);
+        assertEquals(actual, expected, "Parsed version mismatch");
+    }
+
+    private static Stream<Arguments> illegalVersionParams() {
+        return Stream.of(
+                Arguments.of("1.", IllegalArgumentException.class),     // dot without digits
+                Arguments.of("1.2.", IllegalArgumentException.class),   // dot without digits
+                Arguments.of("1.-abc", IllegalArgumentException.class), // dot without digits
+                Arguments.of("1.2.-abc", IllegalArgumentException.class), // dot without digits
+                Arguments.of("", IllegalArgumentException.class),
+                Arguments.of("xaaa", IllegalArgumentException.class),
+                Arguments.of("abc.xyz", IllegalArgumentException.class)
+        );
+    }
+
+    @ParameterizedTest()
+    @MethodSource("illegalVersionParams")
+    public void checkIllegalParse(String verName, Class<Throwable> exception) {
+        try {
+            Version.parse(verName);
+        } catch (Throwable th) {
+            System.err.printf("%s: %s%n", verName, th.getMessage());
+        }
+        assertThrows(exception, () -> Version.parse(verName));
+    }
+
+    private static Stream<Arguments> versionCompare() {
+        return Stream.of(
+                Arguments.of(new Version(2, 1), new Version(2, 1), 0),
+                Arguments.of(new Version(2, 1), new Version(2, 0), 1),
+                Arguments.of(new Version(2, 0), new Version(2, 1), -1),
+                Arguments.of(new Version(3, 3, 1), new Version(3, 3, 1), 0),
+                Arguments.of(new Version(3, 3, 1), new Version(3, 3, 0), 1),
+                Arguments.of(new Version(3, 3, 0), new Version(3, 3, 1), -1)
```

Comment:
This should probably include tests for differing `major` versions as well:
```suggestion
                Arguments.of(new Version(2, 1), new Version(2, 1), 0),
                Arguments.of(new Version(2, 1), new Version(2, 0), +1),
                Arguments.of(new Version(2, 0), new Version(2, 1), -1),
                Arguments.of(new Version(3, 3, 1), new Version(3, 3, 1), 0),
                Arguments.of(new Version(3, 3, 1), new Version(3, 3, 0), +1),
                Arguments.of(new Version(3, 3, 0), new Version(3, 3, 1), -1),
                Arguments.of(new Version(2, 0), new Version(3, 0), -1),
                Arguments.of(new Version(3, 0), new Version(2, 0), +1)
```